src/app/gagestats/gagestats.component.ts

```diff
@@ -46,24 +41,81 @@ export class GagestatsComponent implements OnInit {
     this.loggedInRole = localStorage.getItem('loggedInRole');
     this.title = 'Gage Stats';
     this.timestamp = new Date();
-    this.showStationType = false;
-
     // subscribe to stations subject, which is set in the service's searchStations() function
     this._nssService.Stations.subscribe((s: Array<Station>) => {
         this.selectedStations = s;
-      });
+    });
+    // subscribe to all agencies
+    this._nssService.agencies.subscribe((agencies: Array<Agency>) => {
+      this.agencies = agencies;
+    });
+    // subscribe to all station types
+    this._nssService.stationTypes.subscribe((stationtypes: Array<Agency>) => {
+      this.stationTypes = stationtypes;
+    });
+    //subscribe to page number related information
+    this._nssService.pageResponse.subscribe((pageText: string) => {
+      let response = pageText;
+      this.lastPageNumber = (response.slice(response.indexOf("of") + "of".length));
+      this.lastPageNumber = (this.lastPageNumber.substr(0, this.lastPageNumber.indexOf('.'))); 
+      this.currentPageNumber = (response.slice(response.indexOf("page") + "page".length));
+      this.currentPageNumber = Number(this.currentPageNumber.substr(0, this.currentPageNumber.indexOf('of'))); 
+    });
   }   
 
   showAddStationModal(): void{
     this.gagestatsService.addStation();
   }
 
+  public newPage(event){
+    let pageNumber;
+    if(event.target){
+       pageNumber = (event.target.value);
+    }else{
+       pageNumber = event;
+    }
```

Comment:
again, if you just use `$event.target.value` in the html, you don't have to do all this, you can just use `event` throughout, and can remove this pageNumber variable completely.